### PR TITLE
Username Validation

### PR DESCRIPTION
**Pull request by ritiksp2411**
Fixed the issue for the validation of the Username with length

### CODE REVIEW

**Comment by ritiksp2411** [comment]
@akhilchoudhary2k Please review it once

File: app.js [hunk 1]
```diff
@@ -235,30 +235,40 @@
...
-                        isAdmin: false
-                    }, function(err) {
-                        if (err) console.log(err);
-                    });
-                    passport.authenticate("local")(req, res, function() {
-                        res.redirect("/UserHome");
-                    });
-                }
-            });
+            var trimmedUsername = req.body.username.trim();
```
Comment (akhilchoudhary2k) [inline app.js:238]: Keep the `trim()` intact.
Additionally put a check for cases when there is spaces in between, i.e. username is of >1 words

For example: `abc def` this username should not be allowed
We must throw error to fill the form again.

File: app.js [hunk 2]
```diff
@@ -235,30 +235,40 @@
...
+            if(trimmedUsername.length > 4) {
```
Comment (akhilchoudhary2k) [inline app.js:239]: Make it >=

File: app.js [hunk 3]
```diff
@@ -235,30 +235,40 @@
...
+                            if (err) console.log(err);
+                        });
+                        passport.authenticate("local")(req, res, function() {
+                            res.redirect("/UserHome");
+                        });
+                    }
+                });
+            }
+            else {
+                var message = "username should be more than 4 characters";
```
Comment (akhilchoudhary2k) [inline app.js:267]: username should be ~~more than~~ atleast 4 characters.

**Review by akhilchoudhary2k — CHANGES_REQUESTED** [review]

**Comment by ritiksp2411** [comment]
@akhilchoudhary2k review it once

File: app.js [hunk 4]
```diff
@@ -218,61 +217,62 @@
...
-    schema
-    .is().min(5)                                    // Minimum length 5
-    .is().max(10)                                   // Maximum length 10
-    .has().uppercase()                              // Must have uppercase letters
-    .has().lowercase()                              // Must have lowercase letters
-    .has().digits(1)                                // Must have at least 1 digits
-    .has().not().spaces()                           // Should not have spaces
-    .is().not().oneOf(['Passw0rd', 'Password123']); // Blacklist these values
-    
-    if(schema.validate(req.body.password2)){
```
Comment (akhilchoudhary2k) [inline app.js:231]: Why are you removing password validation ??

**Review by akhilchoudhary2k — CHANGES_REQUESTED** [review]

**Comment by akhilchoudhary2k** [comment]
@ritiksp2411 Seems like you didn't fetch the latest code before raising the PR.

**Pull request closed without merging**